### ISSUE TITLE
XML Responses Fail because AbstractProvider expects arrays

### ISSUE THREAD

**Issue by karllhughes** [JobBrander/jobs-common#21]
Guzzle's XML response handler returns a SimpleXML object with embedded objects in the same format: http://guzzle3.readthedocs.org/http-client/response.html#json-responses

This is a problem for the getJobs() method in the AbstractProvider. Some questions:

- Should a solution to this problem be provided in the common project or should each provider be expected to handle this?
- If we aren't supporting XML, let's be sure to indicate that
- If we do want to support XML, should we try to convert all SimpleXML objects to arrays? Is there a better way to handle this?

**Comment by karllhughes** [comment]
Related to this pull request (now closed): https://github.com/JobBrander/jobs-common/pull/20

**Comment by karllhughes** [comment]
One solution is to do the ol' json_encode/json_decode trick to that $payload variable:

```
    public function getJobs()
    {
        $client = $this->client;
        $verb = strtolower($this->getVerb());
        $url = $this->getUrl();
        $options = $this->getHttpClientOptions();

        $response = $client->{$verb}($url, $options);

        $payload = $response->{$this->getFormat()}();
        $payload = json_decode(json_encode($payload), true);

        $listings = $this->getRawListings($payload);

        return $this->getJobsCollectionFromListings($listings);
    }
```

**Comment by stevenmaguire** [comment]
Closed with #23
